modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -641,7 +641,26 @@ def to_pandas(cls, partitions):
         pandas.DataFrame
             A pandas DataFrame
         """
-        retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
+        retrieved_objects = cls.get_objects_from_partitions(partitions.flatten())
+        if all(
+            map(
+                lambda obj: isinstance(obj, (pandas.DataFrame, pandas.Series)),
+                retrieved_objects,
+            )
+        ):
```

Comment:
takes fewer lines and IMHO easier to understand
```suggestion
        if all(isinstance(obj, (pandas.DataFrame, pandas.Series)) for obj in retrieved_objects):
```